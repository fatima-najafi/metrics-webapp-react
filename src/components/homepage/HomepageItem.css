.item {
  display: flex;
  flex-direction: row;
  height: 100px;
  width: 168px;
  gap: 30px;
  text-align: center;
  padding-left: 20px;
}

.icon {
  height: 60px;
  width: 60px;
  margin-top: 6px;
  margin-left: 8px;
}

.title {
  font-weight: bold;
  font-family: 'Gill Sans', sans-serif;
  font-size: 15px;
  color: #000000b5;
}

/* for desctop version */
@media only screen and (min-width: 768px) {
  .item {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    height: 200px;
    width: 260px;
    border-radius: 10px;
    margin: 20px 10px;
    text-align: center;
  }

  .icon {
    height: 100px;
    width: 100px;
  }

  .title {
    font-weight: bold;
    font-family: 'Gill Sans', sans-serif;
    font-size: 20px;
    margin-top: 20px;
    color: rgb(94, 74, 74);
  }
}
