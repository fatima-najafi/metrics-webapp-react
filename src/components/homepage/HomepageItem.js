import PropTypes from 'prop-types';
import './HomepageItem.css';
import company from '../../images/images (1).jpg';

const HomepageItem = ({ title }) => (
  <div className="container" data-testid="homepageitem">
    <div className="item">
      <img className="icon" src={company} alt="company" />
      <h1 className="title">{title}</h1>
    </div>
  </div>
);

HomepageItem.propTypes = {
  title: PropTypes.string.isRequired,
};

export default HomepageItem;
