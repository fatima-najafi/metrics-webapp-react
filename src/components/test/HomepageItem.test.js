import React from 'react';
import { render } from '@testing-library/react';
import HomepageItem from '../homepage/HomepageItem';

describe('HomepageItem', () => {
  test('renders correctly with the given title', () => {
    const title = 'Example Title';
    const { container } = render(<HomepageItem title={title} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
