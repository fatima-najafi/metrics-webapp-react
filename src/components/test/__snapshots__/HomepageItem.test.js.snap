// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomepageItem renders correctly with the given title 1`] = `
<div
  class="container"
  data-testid="homepageitem"
>
  <div
    class="item"
  >
    <img
      alt="company"
      class="icon"
      src="images (1).jpg"
    />
    <h1
      class="title"
    >
      Example Title
    </h1>
  </div>
</div>
`;
